src/ServiceManagement/RecoveryServices/Commands.RecoveryServices/Service/TestAzureRecoveryServicesVaultUpgrade.cs

```diff
@@ -0,0 +1,110 @@
+    /// </summary>
+    [Cmdlet(VerbsDiagnostic.Test, "AzureRecoveryServicesVaultUpgrade")]
+    public class TestAzureRecoveryServicesVaultUpgrade : RecoveryServicesCmdletBase
+    {
+        #region Parameters
+
+        /// <summary>
+        /// Gets or sets vault type.
+        /// </summary>
+        [Parameter(Mandatory = true)]
+        [ValidateSet(
+            Constants.ASRVaultType,
+            Constants.BackupVault,
+            IgnoreCase = true)]
+        public string ResourceType { get; set; }
+
+        /// <summary>
+        /// Gets or sets vault name.
+        /// </summary>
+        [Parameter(Mandatory = true)]
+        [ValidateNotNullOrEmpty]
+        public string VaultName { get; set; }
+
+        /// <summary>
+        /// Gets or sets location of the vault.
+        /// </summary>
+        [Parameter(Mandatory = true)]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
+
+        /// <summary>
+        /// Gets or sets target resource group.
+        /// </summary>
+        [Parameter(Mandatory = true)]
+        [ValidateNotNullOrEmpty]
+        [Alias("TargetRG", "TargetRGName", "RG")]
+        public string TargetResourceGroupName { get; set; }
+        #endregion Parameters
+
+        /// <summary>
+        /// ProcessRecord of the command.
+        /// </summary>
+        public override void ExecuteCmdlet()
+        {
+            try
+            {
+                this.RecoveryServicesClient.TestVaultUpgradePrerequistes(
+                    this.VaultName,
+                    this.Location,
+                    this.ResourceType,
+                    this.TargetResourceGroupName,
+                    this.Profile.Context.Subscription.Id.ToString());
+
+                this.WriteResponse(Properties.Resources.CheckPrereqSucceded);
+                this.WriteObject(Environment.NewLine);
+            }
+            catch (Exception exception)
+            {
+                ExceptionDetails details =
+                    this.RecoveryServicesClient.HandleVaultUpgradeException(exception);
+                if (!string.IsNullOrEmpty(details.WarningDetails))
+                {
+                    this.WriteWarning(details.WarningDetails);
+                }
+
+                if (!string.IsNullOrEmpty(details.ErrorDetails))
+                {
+                    throw new InvalidOperationException(
+                        Environment.NewLine +
+                        string.Format(
+                            Properties.Resources.ConfirmVaultUpgradePrereqFailed,
+                            Properties.Resources.VaultUpgradeExceptionDetails,
+                            details.ErrorDetails));
+                }
+            }
+        }
+
+        /// <summary>
+        /// Writes content to the screen.
+        /// </summary>
+        /// <param name="contents">Data to be printed on the screen.</param>
+        private void WriteResponse(string contents)
```

Comment:
Removed twice occurrence. 

So the justification of using this is 
Most of our auxportal customers are not even familiar with powershell's way of doing things. Most of them will be using the powershell cmdlets for the first time. So considering that we want to make the ui as close to user as possible(readable). So, I have used the write-object function in a way that it will literally update the customer with the current status of vault upgrade. It will keep updating him after every major point so that he knows what's happening and the thing that migration is not stucked and it's still in progress.

a) **We are not expecting the customers to use the return value of this cmdlet in any way.  
b) It's just a "one time thing" for the customer: once the migration from auxportal to ibiza happens, it's done. People are not going to use it over and over again.**
c) We DO want to show the user all the intermediate steps without even -debug and -verbose. WriteProgress doesn't solbve my purpose as i don't have any progress records. I just have statements.
I saw WriteInformation as well in the base class but it's not implemented both in srevicemanagement and arm code. I am not sure why and whether even this will solve the issue?

If the bold text makes sense, let me know. If you have some better option, we can consider that but please note that "we do want to show the intermediate steps to the client, otherwise the experience would be nothing happening on the powershell screen from past 10-15 mins, which is a really bad experience."